### PR TITLE
KBC-2761 migrate Tables to JSON

### PR DESCRIPTION
**Pull request by martinjunger**
Jira: KBC-2761

Before asking for review make sure that:

## Checklist

- [x] New client method(s) has tests
- [x] Apiary file is updated

## Release

  - [x] I gave the PR a proper label:
    * major (BC break)
    * minor (new feature)
    * patch (backwards compatible fix)
    * no release (just test changes)

### CODE REVIEW

**Comment by martinjunger** [comment]
ℹ️ Rebase

File: src/Keboola/StorageApi/Client.php [hunk 1]
```diff
@@ -1162,7 +1162,7 @@ public function deleteTableAttribute($tableId, $key)
      * @param string $tableId
      * @param string $name
      */
-    public function addTableColumn(string $tableId, string $name, ?array $definition = [], ?string $basetype = null)
+    public function addTableColumn(string $tableId, string $name, ?array $definition = null, ?string $basetype = null)
```
Comment (martinjunger) [inline src/Keboola/StorageApi/Client.php:1165]: Tady to padalo na `Invalid parameters - definition: This field was not expected.`
Protoze kdyz se zavolala `addTableColumn('foo', 'bar')`, tak se `definition` nastavila na `[]`, ale dole se kontroluje `!== null`. Tzn. ze se posilal parametr `definition`.

Z nejakyho duvodu, ve **form-data** se parametr s prazdnym polem vynecha, takze se poslalo: `name=bar`.
Ale pro **JSON** se poslou kompletni data: `{name: bar, definition: []}`. 🤷

**Comment by martinjunger** [comment]
[Tag](https://dev.azure.com/keboola-dev/Connection%20build/_build/results?buildId=25879&view=results) prosel 🟢

**Review by jirkasemmler — APPROVED** [review]
cajk, jen by bylo super mit uz nachystane issues v jire se zalinkovanyma TODO co se tady pridaly, aby byl nejaky jasne definovany krok, kdy se ta TODO odstrani

**Comment by martinjunger** [comment]
> cajk, jen by bylo super mit uz nachystane issues v jire se zalinkovanyma TODO co se tady pridaly, aby byl nejaky jasne definovany krok, kdy se ta TODO odstrani

Tady https://keboola.atlassian.net/browse/KBC-2941 a tady https://keboola.atlassian.net/browse/KBC-2945 jako soucast stejneho epicu.

**Pull request merged**